### ISSUE TITLE
Implement getFormat of DEN

### ISSUE THREAD

**Issue by umichyujl** [bkochuna/ners570f22-SpMV#37]
## Background
The library needs the original dense format to convert various formats.

## Tasks
Make a getFormat method for DEN, which includes numbers of rows and columns and the values.

## Definition of Done
Return the format of the dense matrix

**Comment by SoumiKDaS1701** [comment]
Hey, your getformat() implementation in DEN is showing compilation error, you should fix that before merging DEN to main.

**Comment by SoumiKDaS1701** [comment]
Take a look at the errors here, 
https://github.com/bkochuna/ners570f22-SpMV/actions/runs/3422147837/jobs/5699190838

**Comment by umichyujl** [comment]
> Hey, your getformat() implementation in DEN is showing compilation error, you should fix that before merging DEN to main.

Hello! I am trying to figure it out because literally I wrote the same thing in JDS. For the errors here, I think we need to merge all formats of the .hpp files to get it complied, sorry for merging it into main (I thought the function should be right here).

**Comment by umichyujl** [comment]
> Take a look at the errors here, https://github.com/bkochuna/ners570f22-SpMV/actions/runs/3422147837/jobs/5699190838

I am trying to fix some bugs here.

**Comment by SoumiKDaS1701** [comment]
It is better than previous, but still isn't free of errors. The issue is in declaring Sparsematrix_JDS/ELL/CSR in your code.

**Comment by umichyujl** [comment]
> It is better than previous, but still isn't free of errors. The issue is in declaring Sparsematrix_JDS/ELL/CSR in your code.

Yeah, basically getFormat() is trying to do some transformations from one to another, so we need to include JDS/ELL/CSR in the header files. I believe when we merge all these .hpp files together, this issue can be solved.

**Comment by umichyujl** [comment]
> It is better than previous, but still isn't free of errors. The issue is in declaring Sparsematrix_JDS/ELL/CSR in your code.

Thanks for pointing this out! I will try to find out if I can solve this.

**Comment by umichyujl** [comment]
Hi @SoumiKDaS1701, sorry to bother you again, what's the bug here? I'm not quite familiar with it. Does it have anything related to my `getFormat` function? Thanks a lot!
https://github.com/bkochuna/ners570f22-SpMV/actions/runs/3422442297/jobs/5699855542
